virttest/utils_net.py

```diff
@@ -1439,13 +1436,19 @@ def get_guest_ip_addr(session, mac_addr, os_type="linux", ip_version="ipv4",
                 return nic_address["ipv4"][-1]
             else:
                 return None
-        else:
-            global_address = [x for x in nic_address["ipv6"]
-                              if not x.lower().startswith("fe80")]
-            if global_address:
-                return global_address[0]
+        elif ip_version == "ipv6":
+            if linklocal:
+                return [x for x in nic_address["ipv6"]
+                        if x.lower().startswith("fe80")][0]
```

Comment:
It would raise a `IndexError` if there is no link local addr.

Reply:
@luckyh updated. Thanks.